File: random/karatsubaMultiplication.java
import java.io.*;
import java.math.*;
class karatsubaMultiplication
{
	public static void main(String args[])throws IOException
	{
		BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
		String n=br.readLine();
		String m=br.readLine();
		karatsubaMultiplication obj=new karatsubaMultiplication();
		String ans=obj.multiply(n,m);
		System.out.println(ans);
	}
	String multiply(String n, String m)
	{
		if(n.length()<=1||m.length()<=1)
		{
			return Integer.toString(Integer.parseInt(n)*Integer.parseInt(m));
		}
		int max=Math.max(n.length(),m.length());
		System.out.println(max);
		int half=max/2;
		System.out.println(half);
		String high1=n.substring(0,half);
		String low1=n.substring(half,n.length());
		String high2=m.substring(0,half);
		String low2=m.substring(half,m.length());
		/*System.out.println(high1);
		System.out.println(low1);
		System.out.println(high2);
		System.out.println(low2);*/
		String z0=multiply(low1,low2);
		String z1=multiply((low1+high1),(low2+high2));
		String z2=multiply(high1,high2);
		return Integer.toString(Integer.parseInt(z2)*(int)Math.pow(10,2*half)+(Integer.parseInt(z1)-Integer.parseInt(z2)-Integer.parseInt(z0))*(int)Math.pow(10,half)+Integer.parseInt(z0));
	}
}

File: random/karatsubaMultiplication.cpp
#include <iostream>
#include <string>
using namespace std;
string multiply(string n,string m)
{
	if(n.length()<=1||m.length()<=1)
		return to_string(atoi(n.c_str())*atoi(m.c_str()));
	long long int m1=max(n.length(),m.length());
	long long int half=m1/2;
	string high1=n.substr(0,half);
	string low1=n.substr(0+half,n.length()-half);
	string high2=m.substr(0,half);
	string low2=m.substr(0+half,m.length()-half);
	/*cout<<high1<<endl;
	cout<<low1<<endl;
	cout<<high2<<endl;
	cout<<low2<<endl;*/
	string z0=
}
int main()
{
	string n,m;
	getline(cin,n);
	getline(cin,m);
	cout<<n<<endl;
	cout<<m<<endl;
	string ans=multiply(n,m);
}